sky/data/storage.py

```diff
@@ -1865,10 +1911,686 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} can consist only of '
+                    'lowercase letters, numbers, and hyphens (-). '
+                    'It must begin and end with a letter or number.')
+
+            # Check for two adjacent hyphens
+            if '--' in name:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} must not contain '
+                    'two adjacent hyphens.')
+
+        else:
+            _raise_no_traceback_name_error('Store name must be specified.')
+        return name
+
+    def initialize(self):
+        """Initializes the AZ Container object on the cloud.
+
+        Initialization involves fetching container if exists, or creating it if
+        it does not. Also, it checks for the existance of the storage account
+        and resource group if provided by the user. If not provided, those are
+        created with a default naming convention.
+
+        Raises:
+            StorageBucketCreateError: If container creation fails or storage
+                account attempted to be created already exists.
+            StorageBucketGetError: If fetching existing container fails.
+            StorageInitError: If general initialization fails.
+            ValueError: When user provided storage account that does not belong
+                to the user through config.yaml.
+        """
+        self.storage_client = data_utils.create_az_client('storage')
+        self.resource_client = data_utils.create_az_client('resource')
+        self.storage_account_name, self.resource_group_name = (
+            self._get_storage_account_and_resource_group())
+
+        # resource_group_name is set to None when using non-sky-managed
+        # public container or private container without authorization.
+        if self.resource_group_name is not None:
+            self.storage_account_key = data_utils.get_az_storage_account_key(
+                self.storage_account_name, self.resource_group_name,
+                self.storage_client, self.resource_client)
+
+        self.container_name, is_new_bucket = self._get_bucket()
+        if self.is_sky_managed is None:
+            # If is_sky_managed is not specified, then this is a new storage
+            # object (i.e., did not exist in global_user_state) and we should
+            # set the is_sky_managed property.
+            # If is_sky_managed is specified, then we take no action.
+            self.is_sky_managed = is_new_bucket
+
+    def _get_storage_account_and_resource_group(
+            self) -> Tuple[str, Optional[str]]:
+        """Get storage account and resource group to be used for AzureBlobStore
+
+        Storage account name and resource group name of the container to be
+        used for AzureBlobStore object is obtained from this function. These
+        are determined by either through the source, config.yaml, or default
+        name.
```

Comment:
Could we list the possible cases in the docstr as well, i.e. infer storage account/use default/...